src/Voron/Data/Containers/Container.cs

```diff
@@ -830,6 +830,9 @@ public static void RemoveFromAllPagesList(LowLevelTransaction llt, in Container
             txState.FreeListRemovals.Add(pageNum);
             txState.Additions.Remove(pageNum);
             txState.Removals.Add(pageNum);
+
+            if (txState.LastFreePageByPageLevelMetadata.TryGetValue(rootContainer.Header.PageLevelMetadata, out var page) && page == pageNum)
```

Comment:
This is wrong, because it assumes that the `PageLevelMetadata` for the root container and the `page` that is being removed are the same.
When they aren't the same, that would be result in the same problem.

Reply:
Does it mean that we need to check all values of `LastFreePageByPageLevelMetadata` dictionary and remove the one that is equal to `pageNum`? Can you provide a valid fix based the added test? And handle also the case you see in your below comment?

Reply:
Pretty much, yes. I'll take care of that later, yes.

Reply:
Thx